File: two_sum_1.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # solution 1
        # prevMap = {}  # val -> index

        # for i, n in enumerate(nums):
        #     diff = target - n
        #     if diff in prevMap:
        #         return [prevMap[diff], i]
        #     prevMap[n] = i

        # solution 2
        # for index, value in enumerate(nums):  # [2, 5, 5, 11]
        #     for i, v in enumerate(nums[index+1:]):  # [5, 5, 11]
        #         if value + v == target:
        #             return [index, index + 1 + i]

        # solution 3
        for i in range(len(nums) - 1):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]


if __name__ == "__main__":
    nums = [2, 5, 5, 11]
    target = 10

    solution = Solution().twoSum(nums, target)
    print(solution)

File: Java/src/com/arrays&hashing/_0001/two_sum_1.java
package com.arrays&hasing._0001;

class TwoSum {
  public int[] twoSum(int[] nums, int target) {
    int[] answer = new int[2];
    for (int i = 0; i < nums.length; i++) {
      for (int j = i + 1; j < nums.length; j++) {
        if (nums[i] + nums[j] == target) {
          answer[0] = i;
          answer[1] = j;
        }
      }
    }
    return answer;
  }

  public static void main(String[] args) {
    int[] nums = { 2, 7, 11, 15 };
    int target = 9;

    TwoSum solution = new TwoSum();
    int[] answer = solution.twoSum(nums, target);

    for (int i = 0; i < answer.length; i++) {
      System.out.println(answer[i]);
    }
  }
}